java/operator/src/main/java/com/instaclustr/cassandra/operator/controller/DataCenterReconciliationController.java

```diff
@@ -0,0 +1,546 @@
+                            "-cp", "/opt/lib/cassandra-sidecar/cassandra-sidecar.jar",
+                            "com.instaclustr.cassandra.sidecar.SidecarRestore",
+                            "-bb", backup.getSpec().getTarget(),
+                            "-c", backup.getMetadata().getLabels().get("cassandra-datacenter"),
+                            "-bi", backup.getMetadata().getLabels().get("cassandra-datacenter"),
+                            "-s", backup.getMetadata().getName(),
+                            "--bs", backup.getSpec().getBackupType(),
+                            "-rs"
+                    ))
+                    .addVolumeMountsItem(new V1VolumeMount()
+                            .name("pod-info")
+                            .mountPath("/etc/podinfo"))
+                    .addVolumeMountsItem(new V1VolumeMount()
+                            .name("config-volume")
+                            .mountPath("/etc/cassandra")
+                    ).addVolumeMountsItem(new V1VolumeMount()
+                            .name("data-volume")
+                            .mountPath("/var/lib/cassandra")
+                    )
+            );
+        }
+
+        final V1beta2StatefulSet statefulSet = new V1beta2StatefulSet()
+                .metadata(dataCenterChildObjectMetadata("%s"))
+                .spec(new V1beta2StatefulSetSpec()
+                        .serviceName("cassandra")
+                        .replicas(dataCenterSpec.getReplicas())
+                        .selector(new V1LabelSelector().matchLabels(dataCenterLabels))
+                        .template(new V1PodTemplateSpec()
+                                .metadata(new V1ObjectMeta().labels(dataCenterLabels))
+                                .spec(podSpec)
+                        )
+                        .addVolumeClaimTemplatesItem(new V1PersistentVolumeClaim()
+                                .metadata(new V1ObjectMeta().name("data-volume"))
+                                .spec(dataCenterSpec.getDataVolumeClaim())
+                        )
+                );
+
+        // if the StatefulSet doesn't exist, create it. Otherwise scale it safely
+        K8sResourceUtils.createOrReplaceResource(
+                () -> appsApi.createNamespacedStatefulSet(statefulSet.getMetadata().getNamespace(), statefulSet, null),
+                () -> replaceStatefulSet(statefulSet)
+        );
+    }
+
+
+    private static void configMapVolumeAddFile(final V1ConfigMap configMap, final V1ConfigMapVolumeSource volumeSource, final String path, final String content) {
+        final String encodedKey = path.replaceAll("\\W", "_");
+
+        configMap.putDataItem(encodedKey, content);
+        volumeSource.addItemsItem(new V1KeyToPath().key(encodedKey).path(path));
+    }
+
+
+    private static String toYamlString(final Object object) {
+        return new Yaml().dump(object);
+    }
+
+    private static final long MB = 1024 * 1024;
```

Comment:
Configurable on operator install?

Reply:
Not sure if redefining the concept of a megabyte would be wise ;) These are just constants to simplify the math in the GC settings calculations.